pkg/checks/external/nodeCheck/main.go

```diff
@@ -0,0 +1,203 @@
+				"Check may or may not complete successfully.")
+		}
+	}
+}
+
+// WaitForKubeProxy first checks the node's age to make sure it's not less than three minutes old. If so, it sleep for
+// one minute and checks if kube proxy is ready and running on the node before running the check.
+func WaitForKubeProxy(client *kubernetes.Clientset, namespace string) {
+
+	khPod, err := getKHPod(client, namespace)
+	if err != nil {
+		// Just log the error and return since this check will not work if there's an error getting the podName
+		log.Errorln("Error getting kuberhealthy pod:", err)
+		return
+	}
+	log.Infoln("Pod is on node:", khPod.Spec.NodeName)
+
+	nodeMinAge := time.Minute * 3
+	nodeNew, err := checkNodeNew(client, khPod, nodeMinAge)
+	if err != nil {
+		// Just log the error and return since this check will not work if there's an error getting the node
+		log.Errorln("Error checking if node is new:", khPod.Spec.NodeName, err)
+		return
+	}
+
+	if nodeNew {
+		sleepDuration := time.Minute
+		log.Infoln("Node is less than", nodeMinAge, "old. Sleeping for", sleepDuration)
+		time.Sleep(sleepDuration)
+
+		log.Infoln("Checking if kube-proxy is running and ready.")
+
+		select {
+		case err := <- waitForKubeProxyReady(client, khPod.Spec.NodeName):
+			if err != nil {
+				// Just log the error.
+				log.Errorln(err)
+			}
+			log.Infoln("Kube proxy is ready. Proceeding to run check.")
+		case <- time.After(time.Duration(time.Minute)):
+			// Just log the error.
+			// TO DO: figure out how to address this. Should the check to skip this run and pass up an error instead?
+			// If kube-proxy isn't ready and running, there's definitely something wrong with the new node coming up.
+			log.Errorln("Timed out checking if kube proxy is ready. Check node:", khPod.Spec.NodeName,
+			"Check may or may not complete successfully.")
+		}
+	}
+}
+
+// getKHPod gets the kuberhealthy pod currently running. The pod is needed to get the pod's node information.
+func getKHPod(client *kubernetes.Clientset, namespace string) (*corev1.Pod, error) {
+
+	var khPod *corev1.Pod
+	podName, err := os.Hostname()
+	if err != nil {
+		// Just log the error and return since this check will not work if there's an error getting the podName
+		log.Errorln("Error getting hostname:", err)
+		return khPod, err
+	}
+
+	log.Infoln("Getting pod:", podName, "in order to get its node information")
+	khPod, err = client.CoreV1().Pods(namespace).Get(podName, v1.GetOptions{})
+	if err != nil {
+		// Just log the error and return since this check will not work if there's an error getting the pod
+		log.Errorln("Error getting pod:", err)
+		return khPod, err
+	}
+	return khPod, err
+}
+
+// checkIfNodeIsNew checks the age of the node the kuberhealthy check pod is on to determine if its "new" or not.
+func checkNodeNew(client *kubernetes.Clientset, khPod *corev1.Pod, nodeMinAge time.Duration) (bool, error) {
```

Comment:
we could export this one so users can use ut, probably... Maybe make the exported function take in a pod name and namespace instead of a `corev1.Pod`?